cmd/bucket-replication.go

```diff
@@ -2675,3 +2688,82 @@ func (p *ReplicationPool) queueMRFHeal(file string) error {
 	}
 	return nil
 }
+
+// load replication stats from disk
+func (p *ReplicationPool) loadStatsFromDisk() (rs map[string]BucketReplicationStats, e error) {
+	data, err := readConfig(p.ctx, p.objLayer, replicationStatsPath)
+	if err != nil && err != errConfigNotFound {
+		return rs, err
+	}
+
+	if len(data) <= 4 {
+		return map[string]BucketReplicationStats{}, nil
+	}
+	// Read repl stats meta header
+	switch binary.LittleEndian.Uint16(data[0:2]) {
+	case replStatsMetaFormat:
+	default:
+		return rs, fmt.Errorf("replication stats: unknown format: %d", binary.LittleEndian.Uint16(data[0:2]))
+	}
+	switch binary.LittleEndian.Uint16(data[2:4]) {
+	case replStatsVersion:
+	default:
+		return rs, fmt.Errorf("replication stats: unknown version: %d", binary.LittleEndian.Uint16(data[2:4]))
+	}
+	ss := BucketStatsMap{}
+	if _, err = ss.UnmarshalMsg(data[4:]); err != nil {
+		return rs, err
+	}
+	rs = make(map[string]BucketReplicationStats, len(ss))
+	for bucket, st := range ss {
+		rs[bucket] = st.ReplicationStats
+	}
+
+	return rs, nil
+}
+
+func (p *ReplicationPool) saveStatsToDisk() {
+	if p == nil || p.objLayer == nil {
+		return
+	}
+	sTimer := time.NewTimer(replStatsSaveInterval)
+	defer sTimer.Stop()
+	for {
+		select {
+		case <-sTimer.C:
+			dui, err := loadDataUsageFromBackend(GlobalContext, newObjectLayerFn())
```

Comment:
If we have 100 nodes, then we will update each three seconds, I guess we can get the number of nodes from globalEndpoints.Hostnames() then sleep longer depending on the number of nodes. Also we need to add a random sleep as well before we start this goroutine

Reply:
Also, there is some race here, what we can possibly do is:
- GetObjectNInfo() returns an ObjectReader (with locked namespace)
- Read the data usage and update it
- PutObject() with NoLock set to true
- Close ObjectReader to free the lock

Reply:
@vadmeste, there shouldn't be any race here - we are just caching the data usage in memory here

Reply:
> @vadmeste, there shouldn't be any race here - we are just caching the data usage in memory here

I am actually not sure how this works yet.. shouldn't we use data usage from the last cluster start ?

Let's have a quick conversation about this, also maybe it would be great to add a comment about how this works exactly.

Reply:
> Is this resolved @poornas @vadmeste ?

Yes